src-tauri/src/models.rs

```diff
@@ -0,0 +1,60 @@
+use std::path::PathBuf;
+use serde::Serialize;
+
+// TestKind represents the kind of test we are dealing with
+// (e.g. a block test, a javascript test, etc.)
+#[derive(Debug, Serialize)]
+pub enum TestKind {
+    Block,
+    Javascript,
+}
+
+// Test represents a single test that can be ran
+// either independently or as part of a suite.
+#[derive(Debug, Serialize)]
+pub struct Test {
+    kind: TestKind,
+    name: String,
+    blocks_file_path: PathBuf,
```

Comment:
After a second look, I think it might make more sense to have the file path as just `file_path` and use it both for blocks & js types, discriminating on the kind. (this because we also need the file path for a pure javascript test)